modules/doi/includes/utilities.inc

```diff
@@ -6,15 +6,40 @@
  */
 
 /**
+ * Get the CrossRef for the given url.
+ *
+ * @param string $url
+ *   A DOI url, or a DOI number with or without the "doi:" prefix. If the $url
+ *   is a CrossRef ID passed in as doi:10.1016/j.tiv.2011.10.017 or
+ *   10.1111/eve.12339 it will not get changed by running it through parse_url.
+ *
```

Comment:
ONE MORE THING yes thanks avessey for the link to https://en.wikipedia.org/wiki/Digital_object_identifier#Nomenclature

We should be sure to use the preferred names of things when referring to DOI-related stuff (i.e., "DOI name" instead of "DOI number", and not making references to "CrossRef ID"s in the comments or this function name so that there's consistency for anyone trying to look up this stuff.